File: Delegate iOS ObjC/Delegate iOS ObjC/GreenViewController.m
#import "GreenViewController.h"

@implementation GreenViewController

-(IBAction)gotoBackAction:(id)sender
{
    NSString *textToSend = sendTxtFld.text;
    [self.delegate send: textToSend];
    
    [self dismissViewControllerAnimated:true
                             completion:nil];
}
@end

File: delegation ios/delegation ios/GreenViewController.swift
import UIKit

class GreenViewController: UIViewController
{
    var delegate : sendingMessageDelegate!
    
    @IBOutlet weak var msgTextField: UITextField!
    @IBAction func goBackAction(_ sender: Any)
    {
        let textToSend = msgTextField.text!
        //we have to do something
        delegate.send(message: textToSend)
        
        self.dismiss(animated: true, completion: nil)
    } 
   
}
